### PR TITLE
Add sign in, sign up and forgot password modules' tests

### PR DESCRIPTION
**Pull request by radyslavkrechet**
There is a trouble with embedded view controller. Forgot password view controller is a parent of link view controller. In tests of first controller I've had a crash because link view controller hasn't view model object (we don't use injection for tests) so I've change link and base controllers code. Let me know if it's not ok and I need to change it with more clean solution.

### CODE REVIEW

File: ShopAppTests/App/Account/Controllers/ChangePasswordViewControllerSpec.swift [hunk 1]
```diff
@@ -93,6 +95,13 @@ class ChangePasswordViewControllerSpec: QuickSpec {
                 disposeBag = DisposeBag()
             }
             
+            it("needs to end editin") {
```
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/Controllers/ChangePasswordViewControllerSpec.swift]: "editing"

File: ShopAppTests/App/Account/ViewModels/SignInViewModelSpec.swift [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            }
+        }
+        
+        describe("when sign in button pressed") {
+            var disposeBag: DisposeBag!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                
+                viewModel.emailText.value = "user@mail.com"
```
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/ViewModels/SignInViewModelSpec.swift]: Move setup values to second context closure in this describe block

File: ShopAppTests/App/Account/ViewModels/SignUpViewModelSpec.swift [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            }
+        }
+        
+        describe("when sign up button pressed") {
+            var disposeBag: DisposeBag!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                
+                viewModel.emailText.value = "user@mail.com"
```
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/ViewModels/SignUpViewModelSpec.swift]: The same with previous comment

File: ShopAppTests/App/Account/Controllers/ChangePasswordViewControllerSpec.swift (hunk 1 above)
Comment (AntonoffEvgeniy) [inline ShopAppTests/App/Account/Controllers/ChangePasswordViewControllerSpec.swift]: "editing"

File: ShopAppTests/App/Account/Controllers/ForgotPasswordViewControllerSpec.swift [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+            }
+            
+            context("if it have at least one symbol in text field view") {
+                it("needs to enable forgot password button") {
+                    viewModelMock.makeForgotPasswordButtonEnabled()
+                    
+                    viewController.viewModel.resetPasswordButtonEnabled
```
Comment (plzen) [inline ShopAppTests/App/Account/Controllers/ForgotPasswordViewControllerSpec.swift]: Can we avoid subscription in tests? Looks like we already have subscription between view controller and view model and just need to verify that button enabled status changed.

**Pull request merged**